src/decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.first(10).join(' ')
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** According to Ruby conventions, classes should be in their files. I strongly recommend you move these 4 classes into 4 different files.